sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -500,6 +499,19 @@ case class WholeStageCodegenRDD(@transient sc: SparkContext, var source: CodeAnd
 
   override def compute(split: Partition,
       context: TaskContext): Iterator[InternalRow] = {
+    try {
+      computeInternal(split, context)
+    } catch {
+      case ex: ClassCastException => {
+        // Any other exception in future might be added here
+        CodeGenerator.refresh(source)
+        computeInternal(split, context)
```

Comment:
What implications does this internal retry have in case the exception comes later down in processing chain? For example if this is a bulk insert/update plan and ClassCastException is something different? Are we expecting the CCE to come at the very start of processing? I guess that should be true for the cases we are attempting to correct, but what about false positives?